File: PopulateDB.java
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.PreparedStatement;
import java.io.FileReader;
import com.opencsv.CSVReader;

public class PopulateDB {
    public static void main(String[] args) {
        Connection c = null;
        try {
            Class.forName("org.hsqldb.jdbc.JDBCDriver");
            c = DriverManager.getConnection("jdbc:hsqldb:file:exam", "SA", "");

            // Populate students
            CSVReader reader = new CSVReader(new FileReader("students.csv"));
            String[] nextLine;
            reader.readNext(); // Skip header
            while ((nextLine = reader.readNext()) != null) {
                PreparedStatement ps = c.prepareStatement("INSERT INTO students (student_id, first_name, last_name) VALUES (?, ?, ?)");
                ps.setInt(1, Integer.parseInt(nextLine[0]));
                ps.setString(2, nextLine[1]);
                ps.setString(3, nextLine[2]);
                ps.executeUpdate();
            }

            // Populate questions
            reader = new CSVReader(new FileReader("questions.csv"));
            reader.readNext(); // Skip header
            while ((nextLine = reader.readNext()) != null) {
                PreparedStatement ps = c.prepareStatement("INSERT INTO questions (question_id, question_text, question_type) VALUES (?, ?, ?)");
                ps.setInt(1, Integer.parseInt(nextLine[0]));
                ps.setString(2, nextLine[1]);
                ps.setString(3, nextLine[2]);
                ps.executeUpdate();
            }

            // Populate answers
            reader = new CSVReader(new FileReader("answers.csv"));
            reader.readNext(); // Skip header
            while ((nextLine = reader.readNext()) != null) {
                PreparedStatement ps = c.prepareStatement("INSERT INTO answers (answer_id, question_id, answer_text, is_correct) VALUES (?, ?, ?, ?)");
                ps.setInt(1, Integer.parseInt(nextLine[0]));
                ps.setInt(2, Integer.parseInt(nextLine[1]));
                ps.setString(3, nextLine[2]);
                ps.setBoolean(4, Boolean.parseBoolean(nextLine[3]));
                ps.executeUpdate();
            }

            c.close();
        } catch (Exception e) {
            System.err.println(e.getClass().getName() + ": " + e.getMessage());
            System.exit(0);
        }
        System.out.println("Data populated successfully");
    }
}

File: populate_db.py
import csv
import random

# Generate student data
with open('students.csv', 'w', newline='') as f:
    writer = csv.writer(f)
    writer.writerow(['student_id', 'first_name', 'last_name'])
    for i in range(1, 101):
        writer.writerow([i, f'Student_{i}', f'Name_{i}'])

# Generate question data
with open('questions.csv', 'w', newline='') as f:
    writer = csv.writer(f)
    writer.writerow(['question_id', 'question_text', 'question_type'])
    for i in range(1, 11):
        writer.writerow([i, f'MCQ Question {i}', 'mcq'])
    for i in range(11, 16):
        writer.writerow([i, f'Essay Question {i}', 'essay'])

# Generate answer data
with open('answers.csv', 'w', newline='') as f:
    writer = csv.writer(f)
    writer.writerow(['answer_id', 'question_id', 'answer_text', 'is_correct'])
    answer_id = 1
    for i in range(1, 11):
        for j in range(1, 5):
            is_correct = (j == 1)
            writer.writerow([answer_id, i, f'Answer {j} for question {i}', is_correct])
            answer_id += 1
